### PR TITLE
Fix security issues and allow listing all spooler directories

### PR DESCRIPTION
**Pull request by mordae**
First, let me thank you for your work on this project. It's going to save us some time on implementing SMS notification system for [National Library of Technology](https://www.techlib.cz/en/) readers.

I have made some changes required for deployment in our setting and will, with my colleagues, write a front-end to present information exposed by this tool's API to our administrators for release as a free software.

Do you think you can merge some of these changes?

### CODE REVIEW

**Comment by lvasiliev** [comment]
@mordae Thanks for submitting this pull request. I'll see your changes.

File: api/__init__.py [hunk 1]
```diff
@@ -31,102 +37,117 @@
...
+def get_sms():
+    return jsonify({})
+
+@app.route('/api/v1.0/sms/<kind>/')
+@auth.login_required
+def list_some_sms(kind):
+    if kind not in KINDS:
+        return not_found(None)
+
+    return jsonify({'message_id': listdir(app.config[kind.upper()])})
```
Comment (lvasiliev) [inline api/__init__.py:121]: If you have more than one user account, other users can read all messages (incoming/sent/failed) using returned message_id(s). You can add conditional into get_some_sms(kind, message_id) like this:

```
if m.get('From') == get(auth.username()):
    return jsonify(dict(m))
else:
    return make_response(jsonify({'error': 'Forbidden'}), 403)
```

File: api/__init__.py [hunk 2]
```diff
@@ -31,102 +37,117 @@
...
-    if username in app.config['USERS']:
-        return app.config['USERS'].get(username)
-    return None
+    return app.config['USERS'].get(username)
 
 @auth.error_handler
 def unauthorized():
     return make_response(jsonify({'error': 'Unauthorized access'}), 401)
 
-@app.route('/api/v1.0/sms', methods=['GET'])
```
Comment (lvasiliev) [inline api/__init__.py:105]: It's url (without authentications) for monitoring system (icinga/nagios/etc).  

```
# Monitoring (nginx.conf)
location /api/v1.0/sms {
    uwsgi_pass 127.0.0.1:5000;
    include uwsgi_params;
}   
```

It's  better to rename (and delete @auth.login_required):

```
@app.route('/api/v1.0/monitoring')
def monitoring():
    return jsonify({'monitoring': 'ok'})
```

**Comment by lvasiliev** [comment]
I like your pull request. What do you say about my comments?

File: api/__init__.py [hunk 3]
```diff
@@ -31,102 +37,117 @@
...
-        if mobile in app.config['MOBILE_PERMS'].get(username):
-            return True
-        else:
-            return None
-    return True
+    if not app.config.has_key('MOBILE_PERMS'):
+        # Number access control disabled.
+        return True
+
+    return mobile in app.config['MOBILE_PERMS'].get(auth.username(), [])
```
Comment (lvasiliev) [inline api/__init__.py:92]: There may be both users with restrictions and without them:

```
def access_mobile(mobile):
    if not app.config.has_key('MOBILE_PERMS'):
        # Number access control disabled.
        return True
    if app.config['MOBILE_PERMS'].get(auth.username()):
        return mobile in app.config['MOBILE_PERMS'].get(auth.username(), [])
    else:
        return True
```

**Comment by mordae** [comment]
I understand and will fix these bugs during the next week. Thanks for your feedback!

**Comment by lvasiliev** [comment]
I merged your pull request and added some features (Flask blueprints, apache style hash for password,  uuid for message_id and other).

**Comment by mordae** [comment]
Awesome! This is probably going to be enough for us. :-)
If you ever come to Prague, feel free to visit [our little library](https://www.techlib.cz/en/2749-about-ntk#tab_photos). If I am still here, I'll buy you a beer.

**Pull request merged**